demos/common/python/models/detr.py

```diff
@@ -0,0 +1,95 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import numpy as np
+
+from .model import Model
+from .utils import Detection, resize_image, load_labels
+
+
+class DETR(Model):
+    def __init__(self, ie, model_path, input_transform, labels=None):
+        super().__init__(ie, model_path, input_transform)
+
+        assert len(self.net.input_info) == 1, "Expected 1 input blob"
+        self.image_blob_name = next(iter(self.net.input_info))
+
+        assert len(self.net.outputs) == 2, "Expected 2 output blobs"
+        self.bboxes_blob_name, self.scores_blob_name = self._parse_outputs()
+
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        assert self.c == 3, "Expected 3-channel input"
+
+    def _parse_outputs(self):
+        bboxes_blob_name = None
+        scores_blob_name = None
+        for name, layer in self.net.outputs.items():
+            if layer.shape[2] == 4:
+                bboxes_blob_name = name
+            elif layer.shape[2] == 92:
```

Comment:
this restriction does not allow you to run model versions trained with another number of classes